src/RageInputDevice.h

```diff
@@ -335,19 +338,20 @@ struct DeviceInput
 	bool bDown;
 
 	RageTimer ts;
+	
+	DeviceInput(InputDevice dev, DeviceButton btn, const std::string& itgDevicePersistentID)
```

Comment:
this constructor is not used anywhere, all other constructors set `uniqueID` to `-1` and it's not even attempted to be set anywhere else

Reply:
The `device_id_handler` function will be called when the constructor is invoked. This is because the constructor explicitly calls `device_id_handler` to initialize the `uniqueID` member. If `device_id_handler` correctly maps persistent ID's to unique ID's, the device will be recognized upon reconnection.